src/server/tiering/op_manager.cc

```diff
@@ -0,0 +1,125 @@
+// Copyright 2022, DragonflyDB authors.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+
+#include "op_manager.h"
+
+#include "base/logging.h"
+#include "core/compact_object.h"
+#include "core/overloaded.h"
+#include "server/tiered_storage.h"
+#include "server/tiering/common.h"
+
+namespace dfly::tiering {
+
+OpManager::~OpManager() {
+  storage_.Close();
+}
+
+Future<std::string> OpManager::Read(const CompactObj& co) {
+  auto [offset, length] = co.GetExternalSlice();
```

Comment:
I know this is still a draft. I just want to make sure I understand correctly, you current implementation does not takes into account that we want to group multiple small objects into a single page blob right?
f.e in here we will need to check which page this offset belongs page = offset & ~4095ULL
and then in the actions we will need to fill copy the string from the read result in offset - page with the relevant length right?

Reply:
Yes, currently it doesn't, I will insert it right here in the future